public/app/features/scenes/dashboard/PanelTimeRange.tsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+
+import { dateMath, getTimeZone, rangeUtil, TimeRange } from '@grafana/data';
+import {
+  SceneComponentProps,
+  sceneGraph,
+  SceneObjectBase,
+  SceneTimeRangeLike,
+  SceneTimeRangeState,
+} from '@grafana/scenes';
+import { TimeZone } from '@grafana/schema';
+import { TimeOverrideResult } from 'app/features/dashboard/utils/panel';
+
+interface PanelTimeRangeState extends SceneTimeRangeState {
+  timeFrom?: string;
+  timeShift?: string;
+}
+
+export class PanelTimeRange extends SceneObjectBase<PanelTimeRangeState> implements SceneTimeRangeLike {
+  public static Component = PanelTimeRangeRenderer;
+
+  private overrideResult: TimeOverrideResult | null = null;
+
+  public constructor(state: Partial<PanelTimeRangeState> = {}) {
+    const from = state.from ?? 'now-6h';
+    const to = state.to ?? 'now';
+    const timeZone = state.timeZone ?? getTimeZone();
+    const value = evaluateTimeRange(from, to, timeZone);
+    super({ from, to, timeZone, value, ...state });
+  }
```

Comment:
One thing the model is lacking is a way to specify what the input state is, what state is readonly and what is private. 

I would be cool to be able to say that PanelTimeRange only has timeFrom and timeShift as the only externally settable states.